merchant-sdk/sdk/src/main/java/net/gini/android/merchant/sdk/GiniMerchant.kt

```diff
@@ -0,0 +1,326 @@
+                            Resource.Success(it.data.toPaymentDetails()) }
+                        }
+                }
+                is ResultWrapper.Error -> {
+                    _paymentFlow.value = ResultWrapper.Error(Throwable("Failed to get document"))
+                }
+                is ResultWrapper.Loading -> {}
+            }
+        }
+    }
+
+    /**
+     * Checks whether the document is payable by fetching the document and its extractions from the
+     * Gini Pay API and verifying that the extractions contain an IBAN.
+     *
+     * @return `true` if the document is payable and `false` otherwise
+     * @throws Exception if there was an error while retrieving the document or the extractions
+     */
+    suspend fun checkIfDocumentIsPayable(documentId: String): Boolean {
+        val extractionsResource = giniHealthAPI.documentManager.getDocument(documentId)
+            .mapSuccess { documentResource ->
+                giniHealthAPI.documentManager.getAllExtractionsWithPolling(documentResource.data)
+            }
+        return when (extractionsResource) {
+            is Resource.Cancelled -> false
+            is Resource.Error -> throw Exception(extractionsResource.exception)
+            is Resource.Success -> extractionsResource.data.compoundExtractions
+                .getPaymentExtraction("iban")?.value?.isNotEmpty() ?: false
+        }
+    }
+
+    fun getFragment(iban: String, recipient: String, amount: String, purpose: String, flowConfiguration: PaymentFlowConfiguration? = null) = PaymentFragment.newInstance(
+        giniMerchant = this,
+        paymentDetails = PaymentDetails(
+            recipient = recipient,
+            iban = iban,
+            purpose = purpose,
+            amount = amount
+        ),
+        paymentFlowConfiguration = flowConfiguration ?: PaymentFlowConfiguration()
+    )
+
+    internal fun emitSDKEvent(state: PaymentState) {
+        when (state) {
+            is PaymentState.Success -> _eventsFlow.tryEmit(
+                MerchantSDKEvents.OnFinishedWithPaymentRequestCreated(
+                    state.paymentRequest.id,
+                    state.paymentProviderName
+                )
+            )
+
+            is PaymentState.Error -> _eventsFlow.tryEmit(MerchantSDKEvents.OnErrorOccurred(state.throwable))
+            PaymentState.Loading -> _eventsFlow.tryEmit(MerchantSDKEvents.OnLoading)
+            PaymentState.NoAction -> _eventsFlow.tryEmit(MerchantSDKEvents.NoAction)
+        }
+    }
+
+    /**
+     * Sets a lifecycle observer to handle state restoration after the system kills the app.
+     *
+     * @param registryOwner The SavedStateRegistryOwner to which the observer is attached.
+     * @param retryScope Should be a scope [setDocumentForReview] would be called in (ex: viewModelScope).
+     */
+    fun setSavedStateRegistryOwner(registryOwner: SavedStateRegistryOwner, retryScope: CoroutineScope) {
```

Comment:
@a-szotyori , not sure if this function is still needed? It's not used anywhere...
If it's needed I can update it to use `paymentDetails` instead of `document`... but not sure if that's the case?
If not, then I can cleanup this class even more.